### PR TITLE
API-202: Create a new attribute group

### PR DESCRIPTION
**Pull request by momoss**
**Description (for Contributor and Core Developer)**

Create an Attribute Group through the API

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | OK
| Changelog updated                 | OK
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/ApiBundle/Controller/AttributeGroupController.php [hunk 1]
```diff
@@ -121,4 +163,104 @@ public function listAction(Request $request)
...
+
+    /**
+     * Get a response with HTTP code when an object is created.
+     *
+     * @param AttributeGroupInterface $attributeGroup
+     * @param int                     $status
+     *
+     * @return Response
+     */
+    protected function getResponse(AttributeGroupInterface $attributeGroup, $status)
```
Comment (LaurentPetard) [inline src/Pim/Bundle/ApiBundle/Controller/AttributeGroupController.php]: If this method is only for creation (according to the comment) it might be better to indicate it in the name

Comment (momoss) [inline src/Pim/Bundle/ApiBundle/Controller/AttributeGroupController.php]: Yes, but this method will be used when an object is updated, comment will be update in another PR, getResponse is good

Comment (ahocquard) [inline src/Pim/Bundle/ApiBundle/Controller/AttributeGroupController.php]: Keep the same phpdoc as the other controller please.
"Get a response with a location header to the created or updated resource." 

And phpdoc in AttributeController (copy past from here I guess ;) ).

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/AttributeGroup/CreateAttributeGroupIntegration.php [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+
+        $expectedContent = [
+            'code'    => 400,
+            'message' => 'Invalid json message received',
+        ];
+
+        $client->request('POST', 'api/rest/v1/attribute-groups', [], [], [], $data);
+        $response = $client->getResponse();
+        $this->assertSame(Response::HTTP_BAD_REQUEST, $response->getStatusCode());
+        $this->assertSame($expectedContent, json_decode($response->getContent(), true));
```
Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/AttributeGroup/CreateAttributeGroupIntegration.php]: do not decode content, test directly the response please

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/AttributeGroup/CreateAttributeGroupIntegration.php [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
```
Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/AttributeGroup/CreateAttributeGroupIntegration.php]: same

File: src/Pim/Component/Catalog/Updater/AttributeGroupUpdater.php [hunk 4]
```diff
@@ -84,11 +129,8 @@ protected function setData($attributeGroup, $field, $data)
             $attributeGroup->setSortOrder($data);
         } elseif ('attributes' == $field) {
             $this->setAttributes($attributeGroup, $data);
-        } elseif ('label' == $field) {
-            foreach ($data as $locale => $label) {
-                $attributeGroup->setLocale($locale);
-                $attributeGroup->setLabel($label);
-            }
+        } elseif ('labels' == $field) {
+            $this->translatableUpdater->update($attributeGroup, $data);
```
Comment (ahocquard) [inline src/Pim/Component/Catalog/Updater/AttributeGroupUpdater.php]: 👍

File: src/Pim/Component/Catalog/Updater/AttributeGroupUpdater.php [hunk 5]
```diff
@@ -63,13 +72,49 @@ public function update($attributeGroup, array $data, array $options = [])
...
+     * @throws UnknownPropertyException
+     */
+    protected function validateDataType($field, $data)
+    {
+        if (in_array($field, ['labels', 'attributes'])) {
+            if (!is_array($data)) {
+                throw InvalidPropertyTypeException::arrayExpected($field, static::class, $data);
+            }
+
+            foreach ($data as $key => $value) {
```
Comment (skeleton) [inline src/Pim/Component/Catalog/Updater/AttributeGroupUpdater.php]: $key is useless

File: src/Pim/Component/Catalog/spec/Updater/AttributeGroupUpdaterSpec.php [hunk 6]
```diff
@@ -133,11 +135,7 @@ function it_does_not_update_attributes_from_the_default_group(
...
 
-        $attributeGroup->setLocale('en_US')->shouldBeCalled();
-        $attributeGroup->setLocale('fr_FR')->shouldBeCalled();
-        $attributeGroup->setLabel('Other')->shouldBeCalled();
-        $attributeGroup->setLabel('Autre')->shouldBeCalled();
-
+        $translatableUpdater->update($attributeGroup, $values['labels'])->shouldBeCalled();
 
         $attributeGroupRepository->findDefaultAttributeGroup()->shouldNotBeCalled();
         $attributeGroup->getAttributes()->shouldNotBeCalled();
```
Comment (ahocquard) [inline src/Pim/Component/Catalog/spec/Updater/AttributeGroupUpdaterSpec.php]: Add spec for the wrong format of the fields (as the family updater for example).

File: src/Pim/Component/Catalog/Updater/AttributeGroupUpdater.php [hunk 7]
```diff
@@ -63,13 +72,49 @@ public function update($attributeGroup, array $data, array $options = [])
...
     /**
+     * Validate the data type of a field.
+     *
+     * @param string $field
+     * @param mixed  $data
+     *
+     * @throws InvalidPropertyTypeException
+     * @throws UnknownPropertyException
+     */
+    protected function validateDataType($field, $data)
```
Comment (ahocquard) [inline src/Pim/Component/Catalog/Updater/AttributeGroupUpdater.php]: not "speced"

**Comment by momoss** [comment]
https://core-ci.akeneo.com/blue/organizations/jenkins/akeneo%2Fpim-community-dev/detail/PR-6157/16/pipeline/
